src/annotation/askui-annotation-step-reporter.ts

```diff
@@ -0,0 +1,80 @@
+import fs from "fs";
+import { Step, Reporter, Annotation, ReporterConfig } from "askui";
+import path from "path";
+import { JSDOM } from 'jsdom';
+
+export enum AnnotationLevel {
+  DISABLED = 'disabled',
+  ON_FAILURE = 'onFailure',
+  ALL = 'all',
+}
+
+export interface AnnotationReporterConfig extends ReporterConfig {
+  annotationLevel: AnnotationLevel;
+  folderPath: string;
+  fileNameTemplate: string;
+}
+
+export class AskUIAnnotationStepReporter implements Reporter {
+
+  reporterConfig: AnnotationReporterConfig = {
+    annotationLevel: AnnotationLevel.DISABLED,
+    folderPath: 'report',
+    fileNameTemplate: '_testStep_annotation',
+  }
```

Comment:
Would directly put it on the root level and not nest it within `reporterConfig` to make it a little less complex because right now it is easy to mix up with `config: ReporterConfig`.